tests/tracking/fluent/test_fluent.py

```diff
@@ -1659,3 +1659,27 @@ def test_subprocess_inherit_registry_uri(tmp_path):
         text=True,
     )
     assert sqlite_uri in stdout
+
+
+def test_end_run_inside_context_manager(tmp_path):
+    sqlite_uri = "sqlite:///{}".format(tmp_path.joinpath("test.db"))
+    mlflow.set_registry_uri(sqlite_uri)
+
+    client = MlflowClient()
+
+    with mlflow.start_run() as parent_run:
+        with mlflow.start_run(nested=True) as child_run:
+            mlflow.end_run("FAILED")
```

Comment:
Instead of using end_run, can we throw an exception?

Reply:
I think no. the bug is triggered when explicitly ending run inside the context